packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -121,15 +121,28 @@ abstract class RenderSliverBoxChildManager {
   /// true without making any assertions.
   bool debugAssertChildListLocked() => true;
 }
+/// Parent data structure used by [RenderSliverWithKeepAliveMixin].
+abstract class KeepAliveParentDataMixin extends ParentData {
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  factory KeepAliveParentDataMixin._() => null;
+
+  /// Whether to keep the child alive even when it is no longer visible.
+  bool keepAlive = false;
+}
+
+// This class exists to dissociate [KeepAlive] from [RenderSliverMultiBoxAdaptor].
+abstract class RenderSliverWithKeepAliveMixin extends RenderSliver{
```

Comment:
i don't really understand the purpose of this class. Can you elaborate?

Reply:
The existence of this class is only intended to separate the KeepAlive from RenderSliverMultiBoxAdaptor. But you're right it doesn't do anything. I should at least add an assert in the `setupParentData` to be sure the child.parentData is inheriting `KeepAliveParentDataMixin`. I don't see what I should do beside this.